src/tokenDefinitions.ts

```diff
@@ -119,15 +121,15 @@ export class DefinitionOfTokenCollectionOnNetwork {
         result.name = tokenName.toString();
         result.ticker = collection;
         result.owner = Address.fromPubkey(owner);
-        result.decimals = properties.NumDecimals.toNumber();
-        result.canPause = properties.CanPause;
-        result.canFreeze = properties.CanFreeze;
-        result.canWipe = properties.CanWipe;
-        result.canUpgrade = properties.CanUpgrade;
-        result.canChangeOwner = properties.CanChangeOwner;
-        result.canAddSpecialRoles = properties.CanAddSpecialRoles;
-        result.canTransferNftCreateRole = properties.CanTransferNFTCreateRole;
-        result.canCreateMultiShard = properties.CanCreateMultiShard;
+        result.decimals = properties.NumDecimals.toNumber() || 0;
```

Comment:
I suppose you want to set a default value in case of null or undefined. I think you should use `??` operator instead of `||`. You can find here a good explanation: https://stackoverflow.com/a/61481631/7745982

Reply:
Indeed, now using `??` operator